mono/metadata/w32handle.c

```diff
@@ -131,6 +130,8 @@ mono_w32handle_unlock_signal_mutex (void)
 void
 mono_w32handle_lock (MonoW32Handle *handle_data)
 {
+	if (!handle_data)
```

Comment:
This cannot transparently fail. Check in the caller that we do not lock a invalid handle.

Reply:
Ignoring null is not failing. Let me see  if it works, there might have been another problem.

Reply:
It fails to do the expected thing, thus breaking assumptions.